src/protocol/src/results.rs

```diff
@@ -75,232 +75,212 @@ impl Into<String> for Severity {
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) enum ConstraintViolationKind {
+#[derive(Debug, PartialEq)]
+pub(crate) enum QueryErrorKind {
+    SchemaAlreadyExists(String),
+    TableAlreadyExists(String),
+    SchemaDoesNotExist(String),
+    TableDoesNotExist(String),
+    ColumnDoesNotExist(Vec<String>),
+    NotSupportedOperation(String),
+    TooManyInsertExpressions,
+
     NumericTypeOutOfRange(PostgreSqlType),
     DataTypeMismatch(PostgreSqlType, String),
     StringTypeLengthMismatch(PostgreSqlType, u64),
 }
 
-/// Represents a constraint violation during query execution
-/// It is separate from QueryError because there can be multiple
-/// of these errors for one QueryError.
-#[derive(Debug, Clone, PartialEq)]
-pub struct ConstraintViolation {
+/// Represents error during query execution
+#[derive(Debug, PartialEq)]
+pub(crate) struct QueryErrorInner {
     severity: Severity,
     code: String,
-    kind: ConstraintViolationKind,
+    kind: QueryErrorKind,
 }
 
-impl ConstraintViolation {
-    /// sql error code
-    pub fn code(&self) -> Option<String> {
+impl QueryErrorInner {
+    fn code(&self) -> Option<String> {
         Some(self.code.clone())
     }
 
-    /// error severity
-    pub fn severity(&self) -> Option<String> {
+    fn severity(&self) -> Option<String> {
         Some(self.severity.into())
     }
 
-    /// error message
-    pub fn message(&self) -> Option<String> {
-        Some(format!("{}", self))
+    fn message(&self) -> Option<String> {
+        Some(format!("{}", self.kind))
     }
+}
 
-    /// numeric out of range constructor
-    pub fn out_of_range(pg_type: PostgreSqlType) -> Self {
-        Self {
-            severity: Severity::Error,
-            code: "22003".to_owned(),
-            kind: ConstraintViolationKind::NumericTypeOutOfRange(pg_type),
-        }
-    }
+/// a container of errors that occured during query execution
+#[derive(Debug, PartialEq)]
+pub struct QueryError {
+    errors: Vec<QueryErrorInner>,
+}
 
-    /// type mismatch constructor
-    pub fn type_mismatch(value: &str, pg_type: PostgreSqlType) -> Self {
-        Self {
-            severity: Severity::Error,
-            code: "2200G".to_owned(),
-            kind: ConstraintViolationKind::DataTypeMismatch(pg_type, value.to_owned()),
-        }
+impl QueryError {
+    pub(crate) fn new(errors: Vec<QueryErrorInner>) -> Self {
+        Self { errors }
     }
 
-    /// length of string types do not match constructor
-    pub fn string_length_mismatch(pg_type: PostgreSqlType, len: u64) -> Self {
-        Self {
-            severity: Severity::Error,
-            code: "22026".to_owned(),
-            kind: ConstraintViolationKind::StringTypeLengthMismatch(pg_type, len),
+    pub(crate) fn into_messages(self) -> Vec<Message> {
```

Comment:
I think a good idea is to implement `Into` trait for `QueryError` and `QueryEvent` created #160

Reply:
I can implement that in this PR or I can leave it as a good first issue for someone else.

Reply:
Let it be done separately